ansible_collection/hpe/nimble/plugins/modules/hpe_nimble_pool.py

```diff
@@ -0,0 +1,351 @@
+      specified pool is then deleted. All the arrays in the pool are assigned to the target pool.
+  name:
+    required: True
+    type: str
+    description:
+    - Name of pool.
+  state:
+    required: True
+    choices:
+    - present
+    - absent
+    - create
+    type: str
+    description:
+    - Choice for pool operation.
+  target:
+    required: False
+    type: str
+    description:
+    - Name of the target pool.
+extends_documentation_fragment: hpe_nimble
+short_description: Manage HPE Nimble Storage pools.
+version_added: 2.9
+'''
+
+EXAMPLES = r'''
+
+# if state is create , then create a pool if not present. Fails if already present.
+# if state is present, then create a pool if not present. Succeed if it already exists.
+- name: Create pool if not present
+  hpe_nimble_pool:
+    hostname: "{{ hostname }}"
+    username: "{{ username }}"
+    password: "{{ password }}"
+    state: "{{ state | default('present') }}"
+    name: "{{ name }}"
+    array_list: "{{ array_list }} "
+    description: "{{ description }}"
+
+- name: Delete pool
+  hpe_nimble_pool:
+    hostname: "{{ hostname }}"
+    username: "{{ username }}"
+    password: "{{ password }}"
+    name: "{{ name }}"
+    state: absent
+
+'''
+RETURN = r'''
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+try:
+    from nimbleclient.v1 import client
+except ImportError:
+    client = None
+import ansible_collections.hpe.nimble.plugins.module_utils.hpe_nimble as utils
+
+
+def create_pool(
+        client_obj,
+        pool_name,
+        **kwargs):
+
+    if utils.is_null_or_empty(pool_name):
+        return (False, False, "Create pool failed as pool name is not present.", {})
+
+    try:
+        pool_resp = client_obj.pools.get(id=None, name=pool_name)
+        if utils.is_null_or_empty(pool_resp):
+            params = utils.remove_null_args(**kwargs)
+            pool_resp = client_obj.pools.create(name=pool_name,
+                                                **params)
+            if pool_resp is not None:
+                return (True, True, "Created pool '%s' successfully." % pool_name, {})
+        else:
+            return (False, False, "Pool '%s' cannot be created as it is already present." % pool_name, {})
+    except Exception as e:
+        return (False, False, "Pool creation failed | %s" % e, {})
+
+
+def update_pool(
+        client_obj,
+        pool_resp,
+        **kwargs):
+
+    if utils.is_null_or_empty(pool_resp):
+        return (False, False, "Update pool failed as pool name is not present.", {})
+
+    try:
+        pool_name = pool_resp.attrs.get("name")
+        changed_attrs_dict, params = utils.remove_unchanged_or_null_args(pool_resp, **kwargs)
+        if changed_attrs_dict.__len__() > 0:
+            pool_resp = client_obj.pools.update(id=pool_resp.attrs.get("id"), **params)
+            return (True, True, "Pool '%s' already present. Modified the following fields:" % pool_name, changed_attrs_dict)
```

Comment:
nit: Please use 'F-strings' format syntax while constructing messages/strings. 

PLEASE UPDATE/FIX THIS IN ALL OTHER PLACES OF THE CODE.